catalog/ui/catalog-ui-search/src/main/webapp/js/ApplicationSetup.js

```diff
@@ -80,6 +81,10 @@ require([
         }
 
         var response = getErrorResponse(event, jqxhr, settings, throwError);
+        if (properties.disableUnknownErrorBox && response.message.substring(0,7) === "Unknown") {
```

Comment:
This code might be able to use a bit of refactoring, but it would be good to do this `response.message.substring` check after the checks in [this section](https://github.com/codice/ddf/pull/2840/files#diff-161be453f112bb95da340c092e44209cR91) as well as the final response that gets displayed can be different from the one returned by `getErrorResponse`

Reply:
This should be independent of what is displayed; the check was done here to immediately return if `disableUnknownErrorBox` is true and it's an unknown error.

Reply:
Yeah, but there could be a `response` that comes back from `getErrorResponse` as an `Unknown Error` and then gets caught in, say, (here)[https://github.com/codice/ddf/pull/2840/files#diff-161be453f112bb95da340c092e44209cR95] and the actual error message that _would_ have been displayed wouldn't have been an "Unknown Error" anymore. So we might be suppressing errors that would have not displayed as an `Unknown Error`.

Reply:
Nice find, moving the logic down